src/Refine/Refine.cpp

```diff
@@ -1074,6 +1191,93 @@ void Refine( const int lv, const UseLBFunc_t UseLBFunc )
 // get the total number of patches at lv+1
    Mis_GetTotalPatchNumber( lv+1 );
 
+// (c1.3.6) convert density/phase to density/real part/imaginary part in hybrid scheme if we switch the level from phase to wave
+#  if ( MODEL == ELBDM && ELBDM_SCHEME == ELBDM_HYBRID )
+   if ( SwitchFinerLevelsToWaveScheme ) {
+
+      //Set corresponding flag
+      for (int level = lv + 1; level <= MAX_LEVEL; ++level) {
```

Comment:
Replace `MAX_LEVEL` with `TOP_LEVEL`. `MAX_LEVEL` is used more like a refinement criterion and thus can change after restarting a simulation. Check other places throughout this PR too.